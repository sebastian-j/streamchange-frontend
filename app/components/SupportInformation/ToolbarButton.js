import styled from 'styled-components';

const ToolbarButton = styled.button`
  background: ${(props) => props.theme.iconButtonBackground};
  border: none;
  color: ${(props) => props.theme.buttonTextColor};
  border-radius: 50%;
  height: 90%;
  padding: 7px 7px;
  margin: 0 15px 0 0;
  text-decoration: none;
  &:hover {
    background-color: ${(props) => props.theme.buttonBackgroundHover};
    color: ${(props) => props.theme.buttonTextColorHover};
  }
`;

export default ToolbarButton;
