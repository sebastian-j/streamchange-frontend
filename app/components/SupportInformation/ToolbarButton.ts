import styled from 'styled-components';

const ToolbarButton = styled.button`
  background: transparent;
  border: none;
  color: ${(props) => props.theme.buttonTextColor};
  border-radius: 50%;
  height: 90%;
  padding: 7px 7px;
  position: relative;
  margin: 0 15px 0 0;
  text-decoration: none;
  &:hover {
    svg.border-hover {
      stroke-dashoffset: 0;
    }
  }
  .border {
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    width: 100%;
    height: 100%;
    margin: auto;
  }
  .border-initial {
    border-radius: 50%;
    border: 1px solid;
    opacity: 0.2;
  }
  svg.border-hover {
    stroke: currentColor;
    stroke-dasharray: 1;
    stroke-dashoffset: 1;
    transition: stroke-dashoffset .25s cubic-bezier(.22,.28,.36,1);
    circle {
      r: calc(50% - .5px);
      transform: rotate(-90deg);
      transform-origin: 50% 50%;
      vector-effect: non-scaling-stroke;
    }
  }
`;

export default ToolbarButton;
