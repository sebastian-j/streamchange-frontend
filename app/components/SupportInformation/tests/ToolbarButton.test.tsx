import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import ToolbarButton from '../ToolbarButton';

const shallowRenderer = createRenderer();

describe('<ToolbarButton />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<ToolbarButton />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
