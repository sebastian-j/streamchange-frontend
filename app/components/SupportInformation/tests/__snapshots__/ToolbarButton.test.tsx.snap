// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolbarButton /> should render and match the snapshot 1`] = `
<button
  className="sc-bdvvtL cCVGgs"
/>
`;
